crypto/privateKey.go

```diff
@@ -0,0 +1,98 @@
+package crypto
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aptos-labs/aptos-go-sdk/internal/util"
+)
+
+// PrivateKeyVariant represents the type of private key
+type PrivateKeyVariant string
+
+const (
+	Ed25519   PrivateKeyVariant = "ed25519"
+	Secp256k1 PrivateKeyVariant = "secp256k1"
+)
```

Comment:
To avoid clashes, please name these `PrivateKeyVariantEd25519`.

The entire package of `crypto` can only use the name once.